src/storage/cache/rpc/message.rs

```diff
@@ -0,0 +1,322 @@
+    /// The buffer of the packet, used to store response data.
+    pub response: Option<FileBlockResponse>,
+    /// The sender to send response back to caller.
+    pub sender: flume::Sender<Result<FileBlockResponse, FileBlockRequest>>,
+}
+
+impl FileBlockPacket {
+    /// Create a new file block packet.
+    pub fn new(
+        block_request: &FileBlockRequest,
+        sender: flume::Sender<Result<FileBlockResponse, FileBlockRequest>>,
+    ) -> Self {
+        Self {
+            // Will be auto set by client
+            seq: 0,
+            // Set operation
+            op: ReqType::FileBlockRequest.to_u8(),
+            request: block_request.clone(),
+            timestamp: 0,
+            response: None,
+            sender,
+        }
+    }
+}
+
+impl Encode for FileBlockPacket {
+    /// Encode the file block packet into a byte buffer.
+    fn encode(&self, buffer: &mut BytesMut) {
+        self.request.encode(buffer);
+    }
+}
+
+impl Packet for FileBlockPacket {
+    /// Get the sequence number of the packet.
+    fn seq(&self) -> u64 {
+        self.seq
+    }
+
+    /// Set the sequence number of the packet.
+    fn set_seq(&mut self, seq: u64) {
+        self.seq = seq;
+    }
+
+    /// Get the operation type of the packet.
+    fn op(&self) -> u8 {
+        self.op
+    }
+
+    /// Set the operation type of the packet.
+    fn set_op(&mut self, op: u8) {
+        self.op = op;
+    }
+
+    /// Get the timestamp of the packet.
+    fn set_timestamp(&mut self, timestamp: u64) {
+        self.timestamp = timestamp;
+    }
+
+    /// get the timestamp of the packet.
+    fn get_timestamp(&self) -> u64 {
+        self.timestamp
+    }
+
+    /// Set response data to current packet
+    fn set_resp_data(&mut self, data: &[u8]) -> Result<(), RpcError> {
+        self.response = Some(FileBlockResponse::decode(data)?);
+        Ok(())
+    }
+
+    /// Get the response data of the packet.
+    fn set_result(&self, status: Result<(), RpcError>) {
+        match status {
+            Ok(()) => {
+                match self.response {
+                    Some(ref resp) => {
+                        let file_block_resp = FileBlockResponse {
+                            file_id: resp.file_id,
+                            block_id: resp.block_id,
+                            block_size: resp.block_size,
+                            status: StatusCode::Success,
+                            data: resp.data.clone(),
+                        };
+                        match self.sender.send(Ok(file_block_resp)) {
+                            Ok(()) => {}
+                            Err(err) => {
+                                error!("Failed to set result: {:?}", err);
+                            }
+                        }
+                    }
+                    None => {
+                        error!("Failed to set result: response is None");
+                        match self.sender.send(Err(self.request.clone())) {
```

Comment:
Note that here use send() which might block,  you should make sure sender is from Unbounded channel ,  but flume does not provide trait to ensure that.

Reply:
Yes, the use of blocking and non-blocking code at runtime in tokio test may affect the logic of the implementation, have switched the relevant to async implementation.